### PR TITLE
Update the REST-based libraries user agent

### LINKED ISSUE #355: Include the version number in the user agent for REST-based libraries
This affects BigQuery and Storage. We default the user agent to "google-cloud-dotnet" - it should be "google-cloud-dotnet/1.0.0-beta01" or whatever.

We can use code similar to https://github.com/googleapis/gax-dotnet/blob/master/src/Google.Api.Gax/UserAgentBuilder.cs to get the version number.

### PR DESCRIPTION
**Pull request by jskeet**
This uses new functionality in GAX, hence the project.json changes.

Fixes #355.

### CODE REVIEW

File: apis/Google.Storage.V1/Google.Storage.V1/StorageClientImpl.cs [hunk 1]
```diff
@@ -34,7 +34,7 @@ namespace Google.Storage.V1
     public sealed partial class StorageClientImpl : StorageClient
     {
         private static readonly object _applicationNameLock = new object();
-        private static string _applicationName = "google-cloud-dotnet";
+        private static string _applicationName = UserAgentHelper.GetDefaultUserAgent(typeof(StorageClient));
```
Comment (chrisdunelm) [inline apis/Google.Storage.V1/Google.Storage.V1/StorageClientImpl.cs]: applicationName doesn't seem quite the right variable name now.

Comment (jskeet) [inline apis/Google.Storage.V1/Google.Storage.V1/StorageClientImpl.cs]: It's the application name as far as the REST library is concerned - it's used as the `BaseClientService.ApplicationName` property, but *we* know it's going to be used as the user agent.

I agree it's not totally ideal, but I'm not sure what would be better, to be honest.

Comment (chrisdunelm) [inline apis/Google.Storage.V1/Google.Storage.V1/StorageClientImpl.cs]: OK, fair enough

**Comment by chrisdunelm** [comment]
LGTM

**Pull request merged**